bee-network/src/peer/info.rs

```diff
@@ -0,0 +1,36 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use libp2p::Multiaddr;
+
+/// Additional information about a peer.
+#[derive(Clone, Debug)]
+pub struct PeerInfo {
+    /// The peer's address.
+    pub address: Multiaddr,
+    /// The peer's alias.
+    pub alias: String,
+    /// The type of relation we have with this peer.
+    pub relation: PeerRelation,
+}
+
+/// Describes the relation with a peer.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum PeerRelation {
+    /// Represents a known peer.
+    Known,
+    /// Represents an unknown peer.
+    Unknown,
+}
+
+impl PeerRelation {
+    /// Returns whether the peer is known.
+    pub fn is_known(&self) -> bool {
+        self.eq(&Self::Known)
```

Comment:
Why not `*self == Self::Known`, out of interest?

Reply:
No particular reason. Can't even remember why I opted to go with less syntactic sugar here :shrug: I guess people like `==` better, so I'll probably change. Should I? Please put a :+1:  or :-1: if you guys have a preferance.

Reply:
`matches!` is probably my favourite thing <3

Reply:
Haha, that's true. Thibault loves the `matches!` macro. His specialty is to do something like `!matches!`.  I like that you can even put guards in there.

Reply:
+1 for `matches!` ✊

Reply:
Done